app/controllers/users_controller.rb

```diff
@@ -16,7 +16,22 @@ def create
     end
   end
 
-  def show
+  def edit
+    if User.find(params[:id]).admin?
```

Comment:
Opportunity to refactor here using current_admin? method in ApplicationController